Makefile

```
@@ -396,4 +400,9 @@ ctags:
 	@find . -name "*\.[chS]" -o -path ./tests -prune -o -path ./check-deps -prune \
 		| xargs ctags --regex-asm='/^(GLOBAL|ENTRY|END)\(([^)]*)\).*/\2/'
 
+c-str-conversion:
+	@for i in ${C_STR_TARGET}; do \
+		sed -e 's/\\/\\\\/g;s/\"/\\"/g;s/$$/\\n\"/;s/^/\"/' $${i} > $${i}.$(C_STR_EXTENSION);\
```

Does `s/^/\"/` part really necessary?

---

(s/^/\"/) is for adding __"__ at the beginning of the line as the pair of __"__  of (s/$/\\n\"/) at the end of the line like *"          <option value=\"1\">1</option>\n"* for string literal concatenation.


Another way add the backslash at the end of the line and add " for the first and last only.
It will work in the sed command
's/\\/\\\\/g;s/\"/\\"/g;s/$/\\n\\/;1s/^/\"/;$ a \"'.

The current commit applied the string literal concatenation.

---

Ah, I see.  Thanks for the explanation.  Looks good now :)

---

I would like to ask a small cosmetic change request as follows.
```suggestion
		sed -e 's#\\#\\\\#g;s#\"#\\"#g;s#$$#\\n\"#;s#^#\"#' $${i} > $${i}.$(C_STR_EXTENSION); \
```
It's to make less confusing by using `#` as a separator instead of `/` because using `/` and `\` together looks confusing.